pkg/provisioner/controller.go

```diff
@@ -291,6 +293,103 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 	// to control access to static buckets via RBAC rules on storage classes.
 	isDynamicProvisioning := isNewBucketByStorageClass(class)
 
+	// Re-Get the claim in order to shorten the race condition where the claim was deleted after provisioning started
+	obc, err = claimForKey(key, c.libClientset)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("OBC was lost before we could provision: %v", err)
+		}
+		return err
+	}
+
+	// Older versions of this library set obc.Spec.BucketName with the name of the bucket generated
+	// from obc.Spec.GenerateBucketName, which causes an error condition when updating to a newer
```

Comment:
AFAIR user do can specify both `bucketname` and `generatebucketname` in yaml file(may not be a valid case but still) and in that case `bucketname` has preference over `generatebucketname` while creating OBC. So this may result in a error

Reply:
This is difficult because later in the code, it is an error condition if both are set, and no provisioning will occur. I could alternately fix this by changing the behavior to be as you say and allow both to be set but to pick `bucketName` as preferred.